docs/spec/governance/state.md

```diff
@@ -134,47 +136,29 @@ For pseudocode purposes, here are the two function we will use to read or write
 
 **Store:**
 * `ProposalProcessingQueue`: A queue `queue[proposalID]` containing all the 
-  `ProposalIDs` of proposals that reached `MinDeposit`. Each round, the oldest 
-  element of `ProposalProcessingQueue` is checked during `BeginBlock` to see if
-  `CurrentTime == VotingStartTime + activeProcedure.VotingPeriod`. If it is, 
-  then the application tallies the votes, compute the votes of each validator and checks if every validator in the valdiator set have voted
-  and, if not, applies `GovernancePenalty`. If the proposal is accepted, deposits are refunded.
-  After that proposal is ejected from `ProposalProcessingQueue` and the next element of the queue is evaluated. 
+  `ProposalIDs` of proposals that reached `MinDeposit`. Each `EndBlock`, all the proposals
+  that have reached the end of their voting period are processed.
+  To process a finished proposal, the application tallies the votes, compute the votes of
+  each validator and checks if every validator in the valdiator set have voted.
+  If the proposal is accepted, deposits are refunded.
 
 And the pseudocode for the `ProposalProcessingQueue`:
 
 ```go
   in EndBlock do 
-    
-    checkProposal()  // First call of the recursive function 
-    
-    
-  // Recursive function. First call in BeginBlock
-  func checkProposal()  
-    proposalID = ProposalProcessingQueue.Peek()
-    if (proposalID == nil)
-      return
 
-    proposal = load(Governance, <proposalID|'proposal'>) // proposal is a const key
     votingProcedure = load(GlobalParams, 'VotingProcedure')
+    
+    for finishedProposalID in GetAllFinishedProposalIDs(block.Time)
+      proposal = load(Governance, <proposalID|'proposal'>) // proposal is a const key
 
-    if (CurrentTime == proposal.VotingStartTime + votingProcedure.VotingPeriod && proposal.CurrentStatus == ProposalStatusActive)
-
-    // End of voting period, tally
-
-      ProposalProcessingQueue.pop()
-      validators = 
-
-
-      Keeper.getAllValidators()
+      validators = Keeper.getAllValidators()
       tmpValMap := map(sdk.Address)ValidatorGovInfo
 
       // Initiate mapping at 0. Validators that remain at 0 at the end of tally will be punished
```

Comment:
This comment (as well as the comment in the code) don't make sense... This comment makes it sound like .Minus being 0 means the validator will be punished, but that's not it.... right?

It seems that Minus should be called DelegatorOverrideShares